CRM/Core/BAO/CustomQuery.php

```diff
@@ -387,20 +387,27 @@ public function where() {
 
               // CRM-14563,CRM-16575 : Special handling of multi-select custom fields
               if ($isSerialized && !empty($value) && !strstr($op, 'NULL') && !strstr($op, 'LIKE')) {
+                $sp = CRM_Core_DAO::VALUE_SEPARATOR;
                 if (strstr($op, 'IN')) {
-                  $value = str_replace(",", "[[:cntrl:]]*|[[:cntrl:]]*", $value);
+                  $value = str_replace(",", "$sp|$sp", $value);
                   $value = str_replace('(', '[[.left-parenthesis.]]', $value);
                   $value = str_replace(')', '[[.right-parenthesis.]]', $value);
                 }
                 $op = (strstr($op, '!') || strstr($op, 'NOT')) ? 'NOT RLIKE' : 'RLIKE';
-                $value = "[[:cntrl:]]*" . $value . "[[:cntrl:]]*";
+                $value = $sp . $value . $sp;
                 if (!$wildcard) {
```

Comment:
@aydun indeed this will fix the bug that you encountered by directly using special DAO separator, but this will fail if the value contain one or more special characters (other than CRM_Core_DAO::VALUE_SEPARATOR). This is PR https://github.com/civicrm/civicrm-core/pull/6409 why [[:cntrl:]]* is used instead of  CRM_Core_DAO::VALUE_SEPARATOR. [[:cntrl:]] holds the place for all Control Characters (including CRM_Core_DAO::VALUE_SEPARATOR) and '*' means 0 or multiple occurrences of special characters. 

@eileenmcnaughton any thought on this ?